### PR TITLE
Fix discharge function on lambdas.

### PR DESCRIPTION
**Pull request by ChristianoBraga**
- Fix discharge for lambdas. A lambda value `< lam X:UplcId T:Term RHO:Map >` should discharge `RHO[X <- undef]` in `T` if `X` in `RHO` or simply `RHO` otherwise.

### CODE REVIEW

File: uplc-discharge.md [hunk 1]
```diff
@@ -16,7 +16,11 @@ module UPLC-DISCHARGE
   
   rule discharge(< con T:TypeConstant C:Constant >) => (con T C)  
 
-  rule discharge(< lam I:UplcId T:Term RHO:Map >) => (lam I dischargeTerm(T, RHO)) 
+  rule discharge(< lam I:UplcId T:Term RHO:Map >) => (lam I dischargeTerm(T, RHO[I <- undef]))
+  requires I in_keys(RHO)
```
Comment (PetarMax) [inline uplc-discharge.md:20]: I would hope that `RHO[I <- undef] ==K RHO` when `notBool(I in_keys(RHO))`, so I think the `requires` here and the other case below might not be needed?

File: uplc-discharge.md [hunk 2]
```diff
@@ -43,7 +47,13 @@ module UPLC-DISCHARGE
 
   rule dischargeTerm((builtin BN:BuiltinName), _) => (builtin BN)
 
-  rule dischargeTerm((lam X:UplcId T:Term), RHO:Map) => (lam X dischargeTerm(T, RHO))
+  rule dischargeTerm((lam X:UplcId T:Term), RHO:Map) =>
+       (lam X dischargeTerm(T, RHO[X <- undef]))
+  requires X in_keys(RHO)
```
Comment (PetarMax) [inline uplc-discharge.md:52]: Same comment as above

**Review by PetarMax — CHANGES_REQUESTED** [review]

File: uplc-discharge.md (hunk 1 above)
Comment (ChristianoBraga) [reply uplc-discharge.md:20]: Yes, map delete is total. I will change it.

File: uplc-discharge.md (hunk 2 above)
Comment (ChristianoBraga) [reply uplc-discharge.md:52]: Yes, map delete is total. I will change it.

**Review by SchmErik — APPROVED** [review]
looks good to me!

**Review by PetarMax — APPROVED** [review]

**Pull request merged**